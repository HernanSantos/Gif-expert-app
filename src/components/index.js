//archivo de barrido

export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifGridItem";