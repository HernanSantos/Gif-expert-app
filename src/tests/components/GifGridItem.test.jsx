import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../components/GifGridItem";


describe('pruebas en gifgriditem', () => {

    const title = "Goku";
    const url = "https://goku.com.ar/";

    test('debe fallar si no paso valores ', () => {
        const {container} = render(<GifGridItem title={title} url = {url}/>)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el url y atl', () => {
        render(<GifGridItem title={title} url = {url}/>);
        //screen.debug();
        //expect(screen.getByRole("img").src).toBe(url);
        //expect(screen.getByRole("img").alt).toBe(title);
        const {src,alt} = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
});