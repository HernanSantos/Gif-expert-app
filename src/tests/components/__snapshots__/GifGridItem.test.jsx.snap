// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gifgriditem debe fallar si no paso valores  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.com.ar/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
